src/main/java/org/example/CalendarManagement/calendarcontroller/EmployeeController.java

```diff
@@ -55,29 +60,23 @@ public ResponseEntity<Response> saveEmployee(@Valid @RequestBody AddEmployeeData
         return new ResponseEntity<Response>(addEmployeeResponse,HttpStatus.CREATED);
     }
     @DeleteMapping("/{id}")
-    public ResponseEntity<Response> removeEmployee(@PathVariable String identity,@RequestParam String findBy){
-
+    public ResponseEntity<Response> removeEmployee(@NotNull @PathVariable(name = "id") String employeeId)  {
         ValidateResponse validateResponseForEmployeeIdentity= null;
-        RemoveEmployeeDataRequest removeEmployeeDataRequest = new RemoveEmployeeDataRequest(identity);
-        if(findBy.equals("id")){
-            validateResponseForEmployeeIdentity = validateEmployeeIdentity.checkEmployeeId(identity);
-
-        }
-        else if(findBy.equals("email")){
-            validateResponseForEmployeeIdentity = validateEmployeeIdentity.checkEmployeeEmail(identity);
-        }
-        else{
-            validateResponseForEmployeeIdentity = new ValidateResponse("find by has to be email or id",false);
-        }
+        RemoveEmployeeDataRequest removeEmployeeDataRequest = new RemoveEmployeeDataRequest(employeeId);
+        validateResponseForEmployeeIdentity = validateEmployeeIdentity.checkEmployeeId(employeeId);
 
         if(!validateResponseForEmployeeIdentity.isValid()){
             return new ResponseEntity<Response>(new Response(validateResponseForEmployeeIdentity.getMessage(),null),HttpStatus.BAD_REQUEST);
         }
         else{
-            Employee deletedEmployee = employeeFacade.removeEmployee(removeEmployeeDataRequest,findBy);
-            return new ResponseEntity<Response>(new Response(null,deletedEmployee),HttpStatus.OK);
+            try {
+                Response deletedEmployeeResponse = employeeFacade.removeEmployee(removeEmployeeDataRequest);
+                return new ResponseEntity<Response>(deletedEmployeeResponse, HttpStatus.OK);
+            }catch (RuntimeException exception)
+            {
+                return  new ResponseEntity<>(new Response("Thrift Error" ,null) , HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
all exceptions we are catching may not be due to thrift, better to send the exception message it self in response ?